core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -1045,6 +1051,22 @@ private int getTTL(MetricId<?> metricId) {
         return ttl;
     }
 
+    /**
+     * We do not want to query past the TTL. Doing so may cause Cassandra to read and load into heap space a large
+     * number of tombstones that could substantial increase read latencies and even lead to OOMEs in Cassandra. See
+     * https://bugzilla.redhat.com/show_bug.cgi?id=1416850 for more info.
+     */
+    private DateRange getDateRange(MetricId<?> metricId, long start, long end) {
+        long ttl = Duration.standardSeconds(getTTL(metricId)).getMillis();
+        if ((System.currentTimeMillis() - end) >= ttl) {
```

Comment:
The code here uses unix timestamps - the number of milliseconds since the unix epoch. Timestamps are not affected by time zones, so this should be fine, right?